File: test/language/optional_members/java/optional_members/OptionalExpressionWithRemovedEnumItemTest.java
package optional_members;

import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.Test;

import optional_members.optional_expression_with_removed_enum_item.Compound;
import zserio.runtime.io.BitBuffer;
import zserio.runtime.io.SerializeUtil;

public class OptionalExpressionWithRemovedEnumItemTest
{
    @Test
    public void writeRead()
    {
        final Compound compound = new Compound(12, new long[] {1, 2});
        final BitBuffer bitBuffer = SerializeUtil.serialize(compound);

        final Compound readCompound = SerializeUtil.deserialize(Compound.class, bitBuffer);
        assertEquals(compound, readCompound);
    }
}

File: test/extensions/language/optional_members/cpp/OptionalExpressionWithRemovedEnumItemTest.cpp
#include "gtest/gtest.h"
#include "optional_members/optional_expression_with_removed_enum_item/Compound.h"
#include "zserio/SerializeUtil.h"

namespace optional_members
{
namespace optional_expression_with_removed_enum_item
{

TEST(OptionalExpressionWithRemovedEnumItemTest, writeRead)
{
    Compound compound(12, {{1, 2}});
    auto bitBuffer = zserio::serialize(compound);

    const Compound readCompound = zserio::deserialize<Compound>(bitBuffer);
    ASSERT_EQ(compound, readCompound);
}

} // namespace optional_expression_with_removed_enum_item
} // namespace optional_members
